external-crates/move/crates/move-compiler/tests/move_2024/parser/mut_keyword.exp

```diff
@@ -7,3 +7,24 @@ error[E01002]: unexpected token
   │                       Unexpected ':'
   │                       Expected an identifier
 
+error[E04007]: incompatible types
+  ┌─ tests/move_2024/parser/mut_keyword.move:5:45
+  │    
+5 │         public fun foo(mut: &mut u64): &mut u64 {
+  │                                        -------- Expected: '&mut u64'
+  │ ╭───────────────────────────────────────────────^
+  │ │ ╭─────────────────────────────────────────────'
+6 │ │ │         mut
+7 │ │ │     }
+  │ ╰─│─────^ Invalid return expression
+  │   ╰─────' Given: '()'
```

Comment:
When a function body fails to parse, we should probably hand back an error expression so that we don't get this sort of user-facing error.

Reply:
I believe we can mitigate this if we handle `mut` in paramater parsing differently, but we can deal with that as a follow-up diff.